src/Features/LanguageServer/Protocol/Features/UnifiedSuggestions/UnifiedSuggestedActions/UnifiedFixAllCodeRefactoringSuggestedAction.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixesAndRefactorings;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.UnifiedSuggestions.UnifiedSuggestedActions;
+
+namespace Microsoft.CodeAnalysis.UnifiedSuggestions
+{
+    /// <summary>
+    /// Similar to FixAllCodeRefactoringSuggestedAction, but in a location that can be used by
+    /// both local Roslyn and LSP.
+    /// </summary>
+    internal class UnifiedFixAllCodeRefactoringSuggestedAction : UnifiedSuggestedAction, IFixAllCodeRefactoringSuggestedAction
+    {
+        public IFixAllState FixAllState { get; }
+
+        public UnifiedFixAllCodeRefactoringSuggestedAction(
+            Workspace workspace,
+            CodeAction codeAction,
+            CodeActionPriority codeActionPriority,
+            IFixAllState fixAllState)
+            : base(workspace, codeAction, codeActionPriority)
+        {
+            FixAllState = fixAllState;
+        }
+    }
+}
```

Comment:
not a complaint, but at some point i really want to understand what's going on here in LSP land.  it feels very weird that we seem to be creating a duplicate hierarchy for them.  not sure why we can't have one hierarchy defined at the workspace/feature level that both LSP and Roslyn can consume easily...

Reply:
I think @allisonchou probably has the most context here.